packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -677,11 +677,15 @@ class _HeroFlight {
       final HeroFlightDirection type = initialManifest.type;
       switch (type) {
         case HeroFlightDirection.pop:
-          return initial.value == 1.0 && initialManifest.isUserGestureTransition
+          return initialManifest.isUserGestureTransition
               // During user gesture transitions, the animation controller isn't
-              // driving the reverse transition, but should still be in a previously
-              // completed stage with the initial value at 1.0.
-              ? initial.status == AnimationStatus.completed
+              // driving the reverse transition, but should have a starting value
+              // approaching 1.0. In cases where the toRoute begane offstage, there
+              // is a slight delay while the toRoute laysout before the hero's
+              // flight begins, so the animation may not begin completed.
+              ? initial.value == 1.0
+                  ? initial.status == AnimationStatus.completed
+                  : initial.status == AnimationStatus.forward
```

Comment:
I can't wrap my head around this. Do you know what is the situation that makes this a forward?

Reply:
ah gotcha, yeah in this case I think status is not important, we should just return true.